File: Python/Civitas.py
from Human import Human

class Civitas(Human):
    # civitas adalah child dari Human
    
    # atribut privat
    __univ = ""
    __email = ""
    __fakultas = ""
    
    # constructor
    def __init__(self, nik="", name="", gender="", univ="", email="", fakultas=""):
        super().__init__(nik, name, gender)
        self.__univ = univ
        self.__email = email
        self.__fakultas = fakultas
    
    # getter setter
    def setUniv(self, univ):
        self.__univ = univ
    
    def getUniv(self):
        return self.__univ
    
    def getFakultas(self):
        return self.__fakultas
    
    def setEmail(self, email):
        self.__email = email
    
    def getEmail(self):
        return self.__email
    
    def getFakultas(self):
        return self.__fakultas

File: CPP/Program/Civitas.cpp
#include "header.hh"

// Constructor
Civitas::Civitas() : Human()
{
    univ = "";
    email = "";
    fakultas = "";
}

Civitas::Civitas(string nik, string name, string gender, string univ, string email, string fakultas) : Human(name, gender, nik)
{
    this->univ = univ;
    this->email = email;
    this->fakultas = fakultas;
}

// Constructor dengan atribut
string Civitas::getUniv()
{
    return univ;
}

// Getter Setter
void Civitas::setUniv(string univ)
{
    this->univ = univ;
}

string Civitas::getEmail()
{
    return email;
}

void Civitas::setEmail(string email)
{
    this->email = email;
}

string Civitas::getFakultas()
{
    return fakultas;
}

void Civitas::setFakultas(string fakultas)
{
    this->fakultas = fakultas;
}

// Destructor
Civitas::~Civitas()
{
}